clang/runtime/dpct-rt/include/dpct/dpl_extras/algorithm.h

```diff
@@ -2414,6 +2414,70 @@ void nontrivial_run_length_encode(ExecutionPolicy &&policy,
   ::std::fill(policy, num_runs, num_runs + 1, ret_dist);
 }
 
+template <typename ExecutionPolicy, typename InputIterator,
+          typename OutputIterator, typename CountIterator,
+          typename UnaryPredicate>
+void partition_if(ExecutionPolicy &&policy, InputIterator input,
+                  OutputIterator output, CountIterator num_true, int num_items,
+                  UnaryPredicate pred, bool reverse_last = true) {
+  dpl::copy(policy, input, input + num_items, output);
+  auto beg_false =
+      dpl::stable_partition(policy, output, output + num_items, pred);
+  auto num_true_items = ::std::distance(output, beg_false);
+  dpl::fill(policy, num_true, num_true + 1, num_true_items);
```

Comment:
Since we are using the `policy` for multiple calls to oneDPL, do we need to wrap this policy uniquely for each call similar to what is done inside of oneDPL when multiple patterns / kernels are launched inside an implementation?

(The same comment for each of these APIs)

Reply:
Yes, that is a good point. This change is necessary.

Reply:
Yes, that is a good point. This change is necessary. Would calling the oneDPL internal `make_wrapped_policy` to do so be acceptable here?

Reply:
`oneapi::dpl::__par_backend_hetero::make_wrapped_policy` is an implementation detail from oneDPL so as a general rule here it should not be used.  For something like this, its not been a problem to duplicate the functionality here in the dpct namespace, with the appropriate comments indicating it and then use that version of the code.

Reply:
I have added this as a utility in the `internal` namespace and wrapped around the policies of the different oneDPL calls.